src/tool.yml

```diff
@@ -8,88 +8,21 @@ tools:
         type: bool
         description: If True, an INFO.txt will be written
         default: true
-  fill_depressions:
-    title: Fill Depressions
-    description: Fill Depressions in a DEM using WhiteboxGIS [fill_depression](https://www.whiteboxgeo.com/manual/wbt_book/available_tools/hydrological_analysis.html#FillDepressions) tool
+    catflow_hillslope_generator:
+    title: CATFLOW Hillslope Generator
+    description: Produces the required Raster (.TIFF) files for running the CATFLOW Hillslope Wizard Tool using [Whitebox GIS](https://www.whiteboxgeo.com/) Platform
     version: 0.1
     parameters:
       dem: 
         type: file
         description: Input Raster DEM file 
-      fix_flats: 
+      clip_extent: 
         type: bool
-        description: flag indicating whether flat areas should have a small gradient applied
-        default: true  
-  aspect:
-    title: Calculate Slope Aspect
-    description: Calculates slope aspect (i.e. slope orientation in degrees clockwise from north) for each grid cell in an input digital elevation model using WhiteboxGIS [Aspect](https://www.whiteboxgeo.com/manual/wbt_book/available_tools/geomorphometric_analysis.html?highlight=aspect#aspect) tool
-    version: 0.1
-    parameters:
-      dem: 
+        description: flag indicating whether the processing area should be clipped based on a provided shapefile
+        default: false        
+      shapefile: 
         type: file
-        description: Input Raster filled DEM file 
-      z_factor: 
+        description: Input shapefile for clipping the processing extent           
+      stream_threshold: 
         type: float
-        description: Optional multiplier for when the vertical and horizontal units are not the same
-        default: None  
```

Comment:
This parameter does not have a default, but is optional, as the tool will set it to 100. The difference between optional and default is a bit confusing. You can add `optional: true` here and add to the description that the threshold is always 100, but you can overwrite this constant